fastchat/serve/gradio_block_arena_vision_anony.py

```diff
@@ -242,6 +246,71 @@ def clear_history(request: gr.Request):
         + [""]
     )
 
+def extract_text_from_pdf(pdf_file_path):
+    """Extract text from a PDF file."""
+    try:
+        with open(pdf_file_path, 'rb') as f:
+            reader = PyPDF2.PdfReader(f)
+            pdf_text = ""
+            for page in reader.pages:
+                pdf_text += page.extract_text()
+            return pdf_text
+    except Exception as e:
+        logger.error(f"Failed to extract text from PDF: {e}")
+        return None
+    
+import os
+import nest_asyncio
+from llama_parse import LlamaParse
+
+nest_asyncio.apply()  # Ensure compatibility with async environments
+
+def pdf_parse(pdf_path):
+    # Set API key, can also be configured in the environment
+    api_key = "LLAMA API"
+
+    # Initialize the LlamaParse object
+    parser = LlamaParse(
+        api_key=api_key, 
+        result_type="markdown",  # Output in Markdown format
+        num_workers=4,           # Number of API calls for batch processing
+        verbose=True,            # Print detailed logs
+        language="en"            # Set language (default is English)
+    )
+
+    # Prepare the output directory and file name
+    output_dir = "outputs"
+    os.makedirs(output_dir, exist_ok=True)
+
+    pdf_name = os.path.splitext(os.path.basename(pdf_path))[0]
+    markdown_file_path = os.path.join(output_dir, f"{pdf_name}.md")
```

Comment:
why do we need this?

Reply:
The markdown_file_path variable is unused, the pdf_name is required for the extra-info parameter for the parser. The documentation is here: https://github.com/run-llama/llama_parse?tab=readme-ov-file#using-with-file-object

Reply:
We don't. It was part my old implementation for calling LlamaParse with subprocess. I will remove.